### PR TITLE
x64: build a Position-Independent Executable (PIE)

### PR DESCRIPTION
**Pull request by Tilka**
This is needed to run on x86_64 OSes that require PIC for strong ASLR, e.g. Android or Hardened Gentoo ([issue 7467](https://code.google.com/p/dolphin-emu/issues/detail?id=7467)).

I feel like I forgot something, can't remember what...

### CODE REVIEW

**Comment by BhaaLseN** [comment]
Mh, other than the broken builders, wasn't there a difference in hardened GCCs that uses a different command line switch for this? Might not be PIE tho, can't remember atm (but it was related to not working on hardened).

**Comment by Sonicadvance1** [comment]
You're most like thinking of the -fno-pie flag that we were looking at.

**Comment by BhaaLseN** [comment]
Sounds like it. Guess this clears that up :)

**Comment by phire** [comment]
I'm very weary about getHeapOp explicitly using a scratch register. Sounds like a good way to introduce bugs. Especially since it only uses the scratch register sometimes at random.

**Comment by Tilka** [comment]
@phire Dropped that function.

**Comment by Tilka** [comment]
Note to self: If we ever want to run multiple CPUs in parallel with the same block cache, we'll get problems with PIE.

**Comment by degasus** [comment]
Anything still missing?

**Comment by Tilka** [comment]
Didn't change anything, just rebased to current master. Should be ready.

File: Source/Core/Core/PowerPC/Jit64/JitAsm.cpp [hunk 1]
```diff
@@ -169,16 +169,17 @@ void Jit64AsmRoutineManager::Generate()
 			TEST(32, R(RSCRATCH), R(RSCRATCH));
 			FixupBranch notfound = J_CC(CC_L);
 			//grab from list and jump to it
-			u64 codePointers = (u64)jit->GetBlockCache()->GetCodePointers();
-			if (codePointers <= INT_MAX)
+			const u8** codePointers =  jit->GetBlockCache()->GetCodePointers();
```
Comment (turbedi) [inline Source/Core/Core/PowerPC/Jit64/JitAsm.cpp]: one space too much?

File: Source/Core/Common/MemoryUtil.cpp [hunk 2]
```diff
@@ -117,42 +75,26 @@ void* AllocateMemoryPages(size_t size)
...
+	if (posix_memalign(&ptr, alignment, size))
 #endif
-
-	// printf("Mapped memory at %p (size %ld)\n", ptr,
-	//	(unsigned long)size);
-
-	if (ptr == nullptr)
 		PanicAlert("Failed to allocate aligned memory");
 
+
```
Comment (turbedi) [inline Source/Core/Common/MemoryUtil.cpp]: unnecessary newline?

**Pull request merged**